lib/src/pgn.dart

```diff
@@ -0,0 +1,1006 @@
+  } else {
+    headers.remove('Variant');
+  }
+
+  final defaultFen = defualtPosition(pos.variant).fen;
+  if (pos.fen != defaultFen) {
+    headers['FEN'] = pos.fen;
+  } else {
+    headers.remove('FEN');
+  }
+}
+
+enum CommentShapeColor {
+  green,
+  red,
+  yellow,
+  blue;
+
+  String get string {
+    switch (this) {
+      case CommentShapeColor.green:
+        return 'Green';
+      case CommentShapeColor.red:
+        return 'Red';
+      case CommentShapeColor.yellow:
+        return 'Yellow';
+      case CommentShapeColor.blue:
+        return 'Blue';
+    }
+  }
+
+  static CommentShapeColor? parseShapeColor(String str) {
+    switch (str) {
+      case 'G':
+        return CommentShapeColor.green;
+      case 'R':
+        return CommentShapeColor.red;
+      case 'Y':
+        return CommentShapeColor.yellow;
+      case 'B':
+        return CommentShapeColor.blue;
+      default:
+        return null;
+    }
+  }
+}
+
+/// A comment shape
+///
+/// Example of a comment shape "[%cal Ra1b2]" with color: Red from:a1 to:b2
+@immutable
+class CommentShape {
+  final CommentShapeColor color;
+  final Square from;
+  final Square to;
+
+  const CommentShape(
+      {required this.color, required this.from, required this.to});
+
+  @override
+  String toString() {
+    return to == from
+        ? '${color.string[0]}${toAlgebraic(to)}'
+        : '${color.string[0]}${toAlgebraic(from)}${toAlgebraic(to)}';
+  }
+}
+
+enum EvalType { pawns, mate }
+
+/// A class containing an evaluation
+/// A Evaluation object can be created used .pawns or .mate contructor
+@immutable
+class Evaluation {
+  final double? pawns;
+  final int? mate;
+  final int? depth;
+  final EvalType evalType;
+
+  /// Constructor to create Evaluation of type pawns
+  const Evaluation.pawns(
+      {required this.pawns,
+      this.depth,
+      this.mate,
+      this.evalType = EvalType.pawns});
+
+  /// Constructor to create Evaluation of type mate
+  const Evaluation.mate(
+      {required this.mate,
+      this.depth,
+      this.pawns,
+      this.evalType = EvalType.mate});
+
+  @override
+  bool operator ==(Object other) =>
+      other is Evaluation &&
+      pawns == other.pawns &&
+      depth == other.depth &&
+      mate == other.mate;
+
+  @override
+  int get hashCode => evalType == EvalType.pawns
+      ? Object.hash(pawns, depth)
+      : Object.hash(mate, depth);
+
+  bool isPawns() => evalType == EvalType.pawns;
+}
+
+/// A comment class
+@immutable
+class Comment {
+  /// Comment string
+  final String? text;
+
+  /// List of comment shapes
+  final List<CommentShape> shapes;
+  final double? clock;
+  final double? emt;
+  final Evaluation? eval;
+
+  const Comment(
+      {this.text, this.shapes = const [], this.clock, this.emt, this.eval});
+
+  @override
+  bool operator ==(Object other) {
+    return other is Comment &&
+        text == other.text &&
+        // shapes == other.shapes &&  List == operator doesnt compare each component of list
```

Comment:
why this code is commented out?

Reply:
This was used in testing to check if two comment objects were equal. But dart doesn't equate the list even if they have the same elements. 
I don't think it's necessary.